src/app/SpectrumConvertApplication.cpp

```diff
@@ -0,0 +1,229 @@
+#include <cstddef>
+#include <cstdio>
+#include "app/tide/abspath.h"
+#include "app/tide/records_to_vector-inl.h"
+
+#include "io/carp.h"
+#include "parameter.h"
+#include "io/SpectrumRecordWriter.h"
+#include "SpectrumConvertApplication.h"
+#include "util/Params.h"
+#include "util/FileUtils.h"
+#include "util/StringUtils.h"
+#include <math.h> 
+#include <map>
+#include "crux_version.h"
+using namespace std;
+#define CHECK(x) GOOGLE_CHECK(x)
+
+SpectrumConvertApplication::SpectrumConvertApplication() {
+/*  remove_index_ = "";
+  spectrum_flag_ = NULL;
+  decoy_num_ = 0;
+  num_range_skipped_ = 0;
+  num_precursors_skipped_ = 0;
+  num_isotopes_skipped_ = 0;
+  num_retained_ = 0;
+  num_spectra_ = 0;
+  num_spectra_searched_ = 0;
+  total_candidate_peptides_ = 0;
+  precursor_window_ = 0;
+  spectrum_min_mz_ = 0; 
+  spectrum_max_mz_ = 0;
+  min_scan_ = 0;
+  max_scan_ = 0;
+  min_peaks_ = 0;
+  min_precursor_charge_ = 0;
+  max_precursor_charge_ = 0;
+  out_tsv_target_  = NULL; // original tide-search output format in tab-delimited text files (txt)
+  out_tsv_decoy_ = NULL;  // original tide-search output format in tab-delimited text files (txt) for the decoy psms only
+  out_mztab_target_ = NULL;      // mzTAB output format
+  out_mztab_decoy_ = NULL;      // mzTAB output format for the decoy psms only
+  out_pin_target_ = NULL;        // pin output format for percolator
+  out_pin_decoy_ = NULL;        // pin output format for percolator for the decoy psms only
+  total_spectra_num_ = 0;       // The total number of spectra searched. This is counted during the spectrum conversion
+*/
+  for (int i = 0; i < NUMBER_LOCK_TYPES; i++) {  // LOCK_TYPES are defined in model/objects.h
+    locks_array_.push_back(new boost::mutex());
+  }
+}
+
+SpectrumConvertApplication::~SpectrumConvertApplication() {
+  for (int i = 0; i < NUMBER_LOCK_TYPES; i++) {
+    delete locks_array_[i];
+  }
+}
+
+int SpectrumConvertApplication::main(int argc, char** argv) {
+  return main(Params::GetStrings("tide spectra file"));
+}
+
+int SpectrumConvertApplication::main(const vector<string>& input_files) {
+
+  carp(CARP_INFO, "Running spectrum-convert...");
+
+  num_threads_ = Params::GetInt("num-threads");
+  if (!Params::Exists("spectrum-outdir")) {
```

Comment:
I think Exists checks if the parameter is defined. But it is defined. I think we need to check if it is different from the default parameter.  Maybe you should use: Params::IsDefault()?